website/docs/docs/build/fill-nulls-advanced.md

```diff
@@ -0,0 +1,125 @@
+
+- Use `fill_nulls_with` for `simple` metrics
+- Use `join_to_timespine` and `fill_nulls_with` together for derived and ratio metrics
+
+### Fill null values for simple metrics
+
+For example, if you'd like to handle days with site visits but no leads, you can use `fill_nulls_with` to set the value for leads to zero on days when there are no conversions.
+
+Let's say you have three metrics:
+
+- `website` `visits`, and `leads`
+- and a derived metric called `leads_to_website_visit` that calculates the ratio of leads to site visits.
+
+On the days when there are no conversions, you can set the value for leads to zero by adding the `fill_nulls_with` parameter to the measure input on the leads metric:
+
+<File name='models/metrics/website.yml'>
+
+```yaml
+metrics:
+  - name: website_visits
+    type: simple
+    type_params:
+      measure:
+        name: bookings
+  - name: leads
+    type: simple
+    type_params:
+      measure:
+        name: bookings
+        fill_nulls_with: 0 # This fills null values with zero
+  - name: leads_to_website_visit
+    type: derived
+    type_params:
+      expr: leads/website_visits
+      metrics:
+        - name: leads
+        - name: website_visits
+```
+
+</File>
+
+The `website_visits` and `leads` metrics have the following data:
+
+| metric_time | website_visits |
+| --- | --- |
+| 2024-01-01 | 50 |
+| 2024-01-02 | 37 |
+| 2024-01-03 | 79 |
+
+
+| metric_time | leads |
+| --- | --- |
+| 2024-01-01 | 5 |
+| 2024-01-03 | 8 |
+* Note that there is no data for `2024-01-02` in the `leads` metric.
+
+Although there are no days without visits, there are days without leads. After applying `fill_nulls_with: 0` to the `leads` metric, querying these metrics together shows zero for leads on days with no conversions:
+
+| metric_time | website_visits | leads |
+| --- | --- | --- |
+| 2024-01-01 | 50 | 5 |
+| 2024-01-02 | 37 | 0 |
+| 2024-01-03 | 79 | 8 |
+
+### Use join_to_timespine to fill null values
+
+To ensure you have a complete set of data for every and daily coverage for metrics calculated from other metrics, you can use `join_to_timespine` to fill null values for `derived` and `ratio` metrics. These metrics are built from other metrics (other calculations), not direct measures (raw data), requiring MetricFlow to have an extra subquery layer to render the metric. The subquery nesting is as follows:
+
+- For `derived` and `ratio` metrics, there are three levels of subquery nesting &mdash; derived or ratio metric → input metrics → input measures.
+- For `simple` and `cumulative` metrics, there are only two levels of subquery nesting &mdash; simple or cumulative metric → input measure.
+
+Because `coalesce` isn't applied to the third, subquery layer for `derived` or `ratio` metrics, this means you could still have nulls in the final result set. This is different to [`conversion` metrics](/docs/build/conversion), which focus on event relationships and don't need daily fills.
+
```

Comment:
might be helpful to provide some more info and examples on why `join_to_timespine` is useful for other metrics since i see you can also add it to cumulative, simple, and conversion. so am trying to understand why we specifically call out derived and ratio, when the other metrics use it too.

Reply:
That's a great point! It's still useful for other metrics if you want data for every date. I can add a section here.

Reply:
amazing, thanks @Jstein77 ! let me know and I'll give it a quick review 🙏

Reply:
Added a call out